Vulkan-Pipeline-Assistant/vulkanrenderer.cpp

```diff
@@ -252,6 +253,10 @@ void VulkanRenderer::CreatePipeline() {
     viewport.height = float(m_window->swapChainImageSize().height());
     viewport.minDepth = 0.0f;
     viewport.maxDepth = 1.0f;
+    //TODO this looks like a memory leak waiting to happen
+    m_config.viewports = new VkViewport[1];
+    m_config.viewports[0] = viewport;
+    m_config.viewportCount = 1;
 
```

Comment:
Remove todo and deal with memory leak. This function is potentially called when the display is reloaded. Make sure you always delete the viewport before this code. I recommend moving this code (along with the VkViewport above it) to below Reload in InitSwapChainResources.

Reply:
That viewport is used again below (viewportState.pViewports = &viewport), I've made a simple change for now to avoid refactoring a lot of the CreatePipeline method